src/BuildExecutable.jl

```diff
@@ -41,16 +41,21 @@ type SysFile
     inference0
 end
 
-function SysFile(exename)
-    buildpath = abspath(dirname(Libdl.dlpath("libjulia")))
+function SysFile(exename, debug=false)
+    buildpath = abspath(dirname(Libdl.dlpath(debug ? "libjulia-debug" : "libjulia")))
     buildfile = joinpath(buildpath, "lib"*exename)
     inference = joinpath(buildpath, "inference")
     inference0 = joinpath(buildpath, "inference0")
     SysFile(buildpath, buildfile, inference, inference0)
 end
 
-function build_executable(exename, script_file, targetdir=nothing, cpu_target="native"; force=false)
-    julia = abspath(joinpath(JULIA_HOME, "julia"))
+function build_executable(exename, script_file, targetdir=nothing, cpu_target="native";
+                          force=false, debug=false)
+    julia = abspath(joinpath(JULIA_HOME, debug ? "julia-debug" : "julia"))
+    if !isfile(julia)
```

Comment:
This is really a bug, but the test failure is not caused by it, I reproduced the failure:

- OS: Linux
- use the latest nightly build julia
- do _NOT_ install the `libgit2` package, make sure there's no `libgit2.so` in the global library search path
- run the test, it fails even on the master branch of BuildExecutable.jl

I found that  (using strace) julia wants to open a file named `libgit2` but not `libgit2.so` under the `targetdir`.